raiden/storage/ulid.py

```diff
@@ -0,0 +1,100 @@
+import random
+from dataclasses import dataclass
+from time import CLOCK_MONOTONIC_RAW, clock_getres, clock_gettime_ns, time_ns
+
+from gevent.lock import Semaphore
+
+from raiden.utils.typing import Any, Generic, Iterable, Iterator, List, Tuple, TypeVar, cast
+
+
+@dataclass(frozen=True, order=True)
+class ULID:
+    """An ULID.
+
+    The generated IDs are lexicographically sorted, however the timestamps are
+    *not* representative of wall time.
+
+    SPEC: https://github.com/ulid/spec
+    """
+
+    identifier: bytes
+
+    def __post_init__(self):
+        assert len(self.identifier) == 16, "id_ must be 16 bytes long"
+
+    @property
+    def timestamp(self):
```

Comment:
great, I was hoping for some "human readable" representation of ulids, for any sort of debugging...